clients/lsp-clangd.el

```diff
@@ -178,6 +178,49 @@ This must be set only once after loading the clang client.")
   :risky t
   :type '(repeat string))
 
+(defun lsp-clangd-find-compile-commands-dir ()
+  "Return the full directory path (or nil) that contain a compile_commands.json.
+
+If multiple compile_commands.json files are
+found in the project directory tree - ask the user to choose."
+  ;; TODO lsp-workspace-root won't find anything at the start of the workspace
+  (let ((candidate-dirs (directory-files-recursively (lsp-workspace-root) "compile_commands.json")))
```

Comment:
Does it make sense to look at only one or 2 levels? If not, what will be the perf for huge code-bases?

Reply:
Yes, I haven't measured perf yet (don't have access to such large codebases), but I thought paying the cost as a one-off should be fine? After the server is instantiated we don't need to run it again, right?

Reply:
If emacs is stuck for 10 seconds this is not fine. I would say anything more than 1 second will result in an issue report, especially considering that this is a regression compared to the case in which you have compile_commands in the root. If we decide to keep it as it is, then I would prefer to have that feature off by default.

Reply:
```suggestion
  (let ((candidate-dirs (directory-files-recursively (lsp-workspace-root) "\\`compile_commands.json\\'")))
```
we don't want to match foo_compile_commands.json5. The \\` forces matching at the beginning (the regex is for basenames) and the \\' at the end (regex anchors).

Reply:
> If emacs is stuck for 10 seconds this is not fine. I would say anything more than 1 second will result in an issue report.

you raise a point that only someone with your maintainer experience would know - not my intention that people flood you with bug reports, so I will happily comply. What's the best way to make this an optional feature that is off by default?

Reply:
alternatively, is there a way to limit directory-files-recursively to go only X levels down from root - i would set it to 2 or 3?

Reply:
So, as per your report, typically we are interested in the root and in the build dir, right? If this is true, I would change this method to:

1. Check if compile_commands.json is in root dir -> return it
2. Get dirs in the root dir and check if any of them contain compile_commands and collect them -> handle matches the way you do now.

I like the fact that we will show info when we cannot find the compile_commands.json

Reply:
There is one more option - not sure if practical(I do not have CPP experience) - can we have a list of predefined locations, e. g. {root}/compile_commands.json, {root}/build/compile_commands.json? Does this make any sense?

Reply:
i don't think we need to set a flag if it's in the root dir - that's where clangd looks by default. 

I have seen build/ directories with subdirectories (release, debug) that have their own compile_commands.json, hence my suggestion for max-depth 3

> I like the fact that we will show info when we cannot find the compile_commands.json

That's my primary UX objective with this, therefore I wanted to make it a warning not an info message